shared/scss/components/_button.scss

```diff
@@ -2,21 +2,29 @@
 *********   Generic Button Component
 *******  
 */
-
 @mixin button(
   $bg: transparent,
+  $color: #fff,
   $radius: 0,
+  $border: none,
   $padding: 1em 2em,
-  $border : none,
   $fontSize: 1em,
-  $transition: none
+  $fontWeight: 400,
+  $transition: none,
+  $hoverBgColor: darken($bg-color, 10%)
 ) {
```

Comment:
Thanks for extending the button component :0 but now it may break some other code that rely on the order of the arguments of it because you have added some new arguments between the old ones and that changes the order they were before, instead I suggest to add the new arguments to the end of the arguments list : 
```scss
@mixin button(
  $bg: transparent,
  $radius: 0,
  $padding: 1em 2em,
  $border: none,
  $fontSize: 1em,
  $transition: none,
// new ones
  $fontWeight: 400,
  $color: #fff,
  $hoverBgColor: color.adjust($bg, $lightness : -10%)
) {
// ...
}
```

Reply:
I appreciate your feedback on the button component. My intention in reorganizing the arguments was to enhance readability and maintain a cleaner structure. I noticed that in the original code, there were inconsistencies, such as spaces between some parameters, which I aimed to address.

Since we are at the beginning of the project, I believe these changes are worthwhile. They provide an opportunity for other developers to update their code to accommodate a more organized approach.

That said, I understand the importance of maintaining compatibility for those relying on the previous argument order.

What do you think we should do? Should we refactor the code according to my suggestions, or should we follow your recommendation to preserve the existing argument order?

Reply:
Yes, absolutely this is cleaner and better and as you said as it is the beginning of the project it is easier to refactor the code, so I agree to keep it in this order and then we have to refactor and update the code that is effected